src/vs/workbench/services/workspaces/common/workspaceTrust.ts

```diff
@@ -127,24 +206,33 @@ export class WorkspaceTrustManagementService extends Disposable implements IWork
 		this.storageService.store(this.storageKey, JSON.stringify(this._trustStateInfo), StorageScope.GLOBAL, StorageTarget.MACHINE);
 		this._onDidChangeTrustedFolders.fire();
 
-		await this.recalculateWorkspaceTrust();
+		await this.updateWorkspaceTrust();
 	}
 
-	protected getWorkspaceUris(): URI[] {
-		const workspaceUris = this.workspaceService.getWorkspace().folders.map(f => f.uri);
-		const workspaceConfiguration = this.workspaceService.getWorkspace().configuration;
+	private getWorkspaceUris(): URI[] {
+		const workspaceUris = this._canonicalWorkspace.folders.map(f => f.uri);
+		const workspaceConfiguration = this._canonicalWorkspace.configuration;
 		if (workspaceConfiguration && !isUntitledWorkspace(workspaceConfiguration, this.environmentService)) {
 			workspaceUris.push(workspaceConfiguration);
 		}
 
 		return workspaceUris;
 	}
 
-	protected async calculateWorkspaceTrust(): Promise<boolean> {
+	private calculateWorkspaceTrust(): boolean {
 		if (!isWorkspaceTrustEnabled(this.configurationService)) {
 			return true;
 		}
 
+		if (!this._initialized) {
```

Comment:
Am I reading the flow correctly, that in the local scenario we always first initialize the workspace trust to `false`?
If so, why is that the case given that in the local scenario we can calculate the trust workspace trust in synchronously.

Reply:
that's right. while in fact we could resolve this in the local case, since we will also have virtual file system resolution to add separately, I decided to make the flow consistent for all cases. the benefit I see here is that we will be able to catch any issues with this flow much earlier because it always occurs instead of being limited to remote cases.